File: RobWork/example/lua/invkin.lua
require("rw")

function openpackage (ns)
  for n,v in pairs(ns) do
    if _G[n] ~= nil then
      print("name clash: " .. n .. " is already defined")
    else
      _G[n] = v
    end
  end
end

openpackage(rw)

local WC_FILE = "/path/to/RobWorkData/devices/serialdev/UR10e_2018/UR10e.xml"

local wc = WorkCellLoaderFactory.load(WC_FILE)
if wc:isNull() then
    error("WorkCell could not be loaded")
end
local device = wc:findSerialDevice("UR10e")
if device:isNull() then
    error("UR10e device could not be found.")
end

local state = wc:getDefaultState()
local solver = ClosedFormIKSolverUR(device:asSerialDeviceCPtr(), state)

local Tdesired = Transform3d(Vector3d(0.2, -0.2, 0.5), EAAd(0, Pi, 0):toRotation3D())
local solutions = solver:solve(Tdesired, state)

print("Inverse Kinematics for " .. device:getName() .. ".")
print(" Base frame: " .. device:getBase():getName())
print(" End/TCP frame: " .. solver:getTCP():getName())
print(" Target Transform: " .. tostring(Tdesired))
print("Found " .. solutions:size() .. " solutions.")
for i= 0,solutions:size()-1,1
do
    print(" " .. tostring(solutions[i]))
end

File: RobWork/example/python/invkin.py
from rw import *

WC_FILE = "/path/to/RobWorkData/devices/serialdev/UR10e_2018/UR10e.xml"

if __name__ == '__main__':
    wc = WorkCellLoaderFactory.load(WC_FILE)
    if wc.isNull():
        raise Exception("WorkCell could not be loaded")
    device = wc.findSerialDevice("UR10e")
    if device.isNull():
        raise Exception("UR10e device could not be found.")

    defState = wc.getDefaultState()
    solver = ClosedFormIKSolverUR(device.asSerialDeviceCPtr(), defState)

    Tdesired = Transform3d(Vector3d(0.2, -0.2, 0.5), EAAd(0, Pi, 0).toRotation3D())
    solutions = solver.solve(Tdesired, defState)

    print("Inverse Kinematics for " + device.getName() + ".")
    print(" Base frame: " + device.getBase().getName())
    print(" End/TCP frame: " + solver.getTCP().getName())
    print(" Target Transform: " + str(Tdesired))
    print("Found " + str(solutions.size()) + " solutions.")
    for solution in solutions:
        print(" " + str(solution))
